### PR TITLE
added symfony/http-client as requirement for composer

### PR DESCRIPTION
**Pull request by hmr-it-jr**
## Description
Added symfony/http-client as requirement for composer because I need this to make sure a new plugin from me will work properly.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [ ] I verified that my code applies to the guidelines (`composer kimai:code-check`)
- [ ] I updated the documentation (see [here](https://github.com/kimai/www.kimai.org/tree/master/_documentation))
- [x] I agree that this code is used in Kimai and will be published under the [MIT license](https://github.com/kevinpapst/kimai2/blob/master/LICENSE)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
         "symfony/flex": "^1.0",
         "symfony/form": "^4.0",
         "symfony/framework-bundle": "^4.0",
+        "symfony/http-client": "4.3.*",
```
Comment (kevinpapst) [inline composer.json]: To strict, must be `"symfony/http-client": "^4.0",`

**Comment by McBochi** [comment]
> do you need that for a public or private plugin? for private plugins, there is another way to achieve that.

I found the same requirement missing when I tried to enable transactional mails via Mailgun.

**Comment by kevinpapst** [comment]
Oh, thats good to know. In that case the library will be included.
Thanks for the feedback!

**Comment by McBochi** [comment]
> Oh, thats good to know. In that case the library will be included.
> Thanks for the feedback!

Not sure why, though... The included `symfony/mailgun-mailer` just states it as development dependency and I assumed this dependency would be resolved by composer during the initial installation. However, when enabling Mailgun as mail service everything stopped working and a manual install fixed it.

**Comment by kevinpapst** [comment]
Closing. I fix it in #1545

**Pull request closed without merging**